network/p2pPeer.go

```diff
@@ -75,8 +75,14 @@ func (c *wsPeerConnP2P) CloseWithMessage([]byte, time.Time) error {
 
 func (c *wsPeerConnP2P) SetReadLimit(int64) {}
 
-func (c *wsPeerConnP2P) CloseWithoutFlush() error {
-	err := c.stream.Close()
+func (c *wsPeerConnP2P) CloseWithoutFlush() (err error) {
+	err = c.stream.Close()
+	defer func() {
+		err0 := c.stream.Conn().Close()
+		if err == nil {
+			err = err0
+		}
+	}()
```

Comment:
Surely it's the job of `c.stream.Close()` to close the internal `Conn()`?  

I'm also worried that the docs for MuxedStream say this about `Close()`
```
	// Close closes the stream.
	//
	// * Any buffered data for writing will be flushed.
```

yet this is being called in `CloseWithoutFlush`?

Reply:
`stream.Close` closes multiplexed stream, not connection itself

Reply:
>yet this is being called in CloseWithoutFlush?

Fair, replaced with `stream.Reset()`.

> I think calling Network().ClosePeer() is the "official" way to do it?

it requires p2p.Service and peerID to be part of wsPeerConnP2P. `ClosePeer` calls `conn.Close()` for all streams. We expect one connection per peer here so `c.stream.Conn().Close()` does the job.